### PR TITLE
Add a syntax SHOW SHARDING TABLE RULES USED ALGORITHM

### LINKED ISSUE #15168: [DistSQL] Add a syntax `SHOW SHARDING TABLE RULES USED ALGORITHM`
Hi community,
this issue is similar to [SHOW RULES USED RESOURCE](https://github.com/apache/shardingsphere/issues/15167)

### Aim
The syntax is as follows:
```sql
SHOW SHARDING TABLE RULES USED ALGORITHM algorithmName [FROM schemaName]
```

### Reference
- #14609
- #15167

### PR DESCRIPTION
**Pull request by leedongyang**
Fixes #15168 

Changes proposed in this pull request:
- add RQL statement
- add test

### CODE REVIEW

**Comment by menghaoranss** [comment]
@leedongyang some it tests execute failed.

**Review by totalo — DISMISSED** [review]
LGTM

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/ShardingTableRulesUsedAlgorithmQueryResultSet.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        data = result.iterator();
+    }
+
+    private void requireResult(final ShowShardingTableRulesUsedAlgorithmStatement statement, final String schemaName, final List<Collection<Object>> result,
+                               final ShardingRuleConfiguration shardingRuleConfiguration) {
+        if (!statement.getAlgorithmName().isPresent()) {
+            return;
+        }
+        shardingRuleConfiguration.getTables().forEach(each -> {
+            if (null != each.getTableShardingStrategy() && statement.getAlgorithmName().get().equals(each.getTableShardingStrategy().getShardingAlgorithmName())) {
```
Comment (RaigorJiang) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/query/ShardingTableRulesUsedAlgorithmQueryResultSet.java:55]: Hi @leedongyang , `databaseShardingStrategy` also needs to be checked here.

**Review by RaigorJiang — APPROVED** [review]

**Comment by RaigorJiang** [comment]
Thank you @leedongyang , well done!

**Pull request merged**